src/torchpme/calculators/p3m.py

```diff
@@ -68,13 +62,23 @@ def __init__(
             potential=potential,
             full_neighbor_list=full_neighbor_list,
             prefactor=prefactor,
-            dtype=dtype,
-            device=device,
         )
 
+        if potential.smearing is None:
+            raise ValueError(
+                "Must specify smearing to use a potential with P3MCalculator"
+            )
+
+        cell = torch.eye(
+            3,
+            device=self.potential.smearing.device,
+            dtype=self.potential.smearing.dtype,
+        )
+        ns_mesh = torch.ones(3, dtype=int, device=cell.device)
+
```

Comment:
When I apply to `.to` to the class, is this correctly passed to `self.kspace_filter` and `self.mesh_interpolator`?